Display average rating and total ratings in UI

AlperenSamurlu opened the pull request:
https://github.com/user-attachments/assets/d806e8a7-3dde-4cf1-bff8-e84e5c8bac28



Added 'Average Rating' and 'Total Ratings' labels to localization and updated the rating component to show average rating and total ratings in the modal and main view. Calculated average rating in the view component and passed it to the view model. Removed undo rating functionality and related UI/JS code for a simplified rating experience.

### Description

Resolves #23615

AlperenSamurlu committed:
Display average rating and total ratings in UI

Added 'Average Rating' and 'Total Ratings' labels to localization and updated the rating component to show average rating and total ratings in the modal and main view. Calculated average rating in the view component and passed it to the view model. Removed undo rating functionality and related UI/JS code for a simplified rating experience.

yagmurcelk reviewed (changes requested):
- [ ] When we update the ratings, the star types in the rating details are not reflected correctly.

[screen-capture (25).webm](https://github.com/user-attachments/assets/3f573422-d2ff-40e3-b6a3-1c7375a20c98)

AlperenSamurlu committed:
Make rating modal IDs unique per entity

Generates unique modal and label IDs for the rating detail modal based on entity type and ID, preventing conflicts when multiple rating components are rendered on the same page.
